### PR TITLE
Metrics improvements

### PR DESCRIPTION
**Pull request by Eadword**
OPT-303

### CODE REVIEW

**Comment by Eadword** [comment]
@dbeal-wiser @dkvasnicka With the exception of this not actually getting the http path when I test it on CCS, it is ready. Not sure what to do about that issue at this time.

File: lib/http_server/server.js [hunk 1]
```diff
@@ -2,11 +2,23 @@ const Koa = require('koa')
...
 module.exports = class {
-  constructor(app, container, config, log) {
+  constructor(app, container, config, log, prometheus) {
     this.app = app
     this.log = log
     this.container = container
     this.config = config
+    this.metrics = {
+      responseTime: new prometheus.Histogram({
+        name: `${prometheus.prefix}http_resonse_time`,
```
Comment (dkvasnicka) [inline lib/http_server/server.js]: Typo.

File: lib/http_server/server.js [hunk 2]
```diff
@@ -30,6 +42,23 @@ module.exports = class {
...
 
+    this.koa.use(async (ctx, next) => {
+      const path = ctx._matchedRoute || ctx._matchedRouteName
+      const method = ctx.method
+      const timer = this.metrics.responseTime.startTimer({path, method})
+      await next()
+      timer()
+
+      let sentiment
+      if (ctx.status < 200) sentiment = '1xx'
```
Comment (dkvasnicka) [inline lib/http_server/server.js]: Seems a little bit too chatty to me.
I would suggest this:
```const sentiment = `${Math.floor(ctx.status / 100)}xx` ```

**Comment by Eadword** [comment]
OPT-303 opened to track HTTP path work required

**Pull request merged**